avocado/core/job.py

```diff
@@ -89,7 +89,9 @@ def __init__(self, args=None):
         if unique_id is None:
             unique_id = job_id.create_unique_job_id()
         self.unique_id = unique_id
-        self.logdir = None
+        self._setup_job_results()
+        if not (self.standalone or getattr(self.args, "dry_run", False)):
+            self._update_latest_link()
```

Comment:
Yes, I thought about that and I think it should be created during `__init__` for the same reasons you described. Basically the job is not initialized until it has results dir.